### PR TITLE
[Fixes] Fix DataProtection Backup Vaults

### PR DESCRIPTION
**Pull request by JPEasier**
# Description

#2160 
 Due to a change in Azure API one of the properties of backupVault are invalid.
 `StorageSettings.datastoreType` value was `SnapshotStore` and now `OperationalStore`.

- Update API version.
- Update readme.md.

## Pipeline references

| Pipeline |
| - |
| [DataProtection: BackupVaults](https://github.com/Azure/ResourceModules/actions/workflows/ms.dataprotection.backupvaults.yml) |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.DataProtection/backupVaults/readme.md [hunk 1]
```diff
@@ -14,10 +14,10 @@ This module deploys DataProtection BackupVaults.
 
 | Resource Type | API Version |
 | :-- | :-- |
-| `Microsoft.Authorization/locks` | [2017-04-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Authorization/2017-04-01/locks) |
+| `Microsoft.Authorization/locks` | [2017-04-01](https://docs.microsoft.com/en-us/azure/templates) |
```
Comment (ahmadabdalla) [inline modules/Microsoft.DataProtection/backupVaults/readme.md:17]: @AlexanderSehr i think this URL is impacted by the recent doco error that was raised internally. Should we wait for that to be fixed before merging? As we might have to regenerate docs later

Comment (AlexanderSehr) [reply modules/Microsoft.DataProtection/backupVaults/readme.md:17]: We should and it is. @JPEasier please re-run the generation. The link should be back now :)

Comment (JPEasier) [reply modules/Microsoft.DataProtection/backupVaults/readme.md]: @ahmadabdalla, @AlexanderSehr it's updated now ;)

File: modules/Microsoft.DataProtection/backupVaults/readme.md [hunk 2]
```diff
@@ -365,55 +365,8 @@ module backupVaults './Microsoft.DataProtection/backupVaults/deploy.bicep' = {
...
-                    dataStoreType: 'OperationalStore'
-                    objectType: 'DataStoreInfoBase'
-                  }
-                  targetDataStoreCopySettings: []
-                }
-              ]
-              name: 'Default'
-              objectType: 'AzureRetentionRule'
-            }
+            'System.Management.Automation.OrderedHashtable'
```
Comment (eriqua) [inline modules/Microsoft.DataProtection/backupVaults/readme.md:368]: @AlexanderSehr

Comment (AlexanderSehr) [reply modules/Microsoft.DataProtection/backupVaults/readme.md:368]: Jap, this is wrong ;)

Comment (AlexanderSehr) [reply modules/Microsoft.DataProtection/backupVaults/readme.md:368]: @JPEasier how did you generate this?

Comment (AlexanderSehr) [reply modules/Microsoft.DataProtection/backupVaults/readme.md:368]: Once #2315 is merged, please merge latest main into your branch and re-run the `Set-ModuleReadMe` script

Comment (AlexanderSehr) [reply modules/Microsoft.DataProtection/backupVaults/readme.md:368]: @JPEasier , you're good to go :)

Comment (JPEasier) [reply modules/Microsoft.DataProtection/backupVaults/readme.md]: @eriqua, @AlexanderSehr , the readme is updated.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**